### DISCUSSION TITLE
zone files with dynamic records

### DISCUSSION THREAD

**Discussion by ctnadovich** [samboy/MaraDNS#114] [Q&A] [answered]
I scoured the documentation, but I don't see anywhere noted how to tell maradns that a zone file has changed. I assume the only way is to completely restart maradns, but I'm wondering if there is another way (eg send a HUP, rebuild some database, etc...) that might be faster and more atomic. The reason I'm asking is I have some records in my zone that are dynamic. I would have an external process rewrite the zone file and then do what? Restart maradns?  I know I could figure this out by experimenting and reading the code, but this seems like something that should be explained in either the zone file man page or the maradns man page.

**Comment by samboy** [comment, marked as answer]
Well, the answer, in fact, is in the documentation, but not in the man page, FAQ, or other obvious place: https://github.com/samboy/MaraDNS/blob/master/doc/en/text/dnsslave.txt

_Note also that maradns must be restarted to reload the updated zone files._

I should probably add a note to the maradns or mararc man page.

(Reloadable zone files would probably require using a b-tree instead of a hash for the internal representation of records; the advantage of a single hash is that it’s *faster*)

**Comment by samboy** [comment]
I have updated MaraDNS’s documentation to note this in the maradns and mararc man pages.

For people who wish this functionality, NSD or KnotDNS might (or might not, I haven’t checked) work out for you:

https://www.nlnetlabs.nl/projects/nsd/about/

https://www.knot-dns.cz/

**Comment by samboy** [comment]
Locking discussion to stop spam bots.
